pkg/placement/placement.go

```diff
@@ -43,34 +43,56 @@ import (
 
 const (
 	waitBeforeTrackingPlacements = 5 * time.Second
-	PlacementKind                = "Placement"
 	KSFinalizer                  = "placement.kubestellar.io/kscontroller"
 )
 
 // Handle placement as follows:
-//  1. requeue all objects to account for changes in placement
-//  2. handle finalizers and deletion of objects associated with the placement
+//
+//  1. if placement is not being deleted:
+//
+//     - update the (where) resolution of the placement and queue the
+//     associated placement decision for syncing.
+//
+//     - requeue workload objects to account for changes in placement
+//
+//     otherwise:
+//
+//     - delete the resolution of the placement.
+//
+//  2. handle finalizers and deletion of objects associated with the placement.
+//
 //  3. for updates on label selectors, re-evaluate if existing objects should be removed
 //     from clusters.
 func (c *Controller) handlePlacement(obj runtime.Object) error {
-	placement := obj.DeepCopyObject()
+	placement, err := runtimeObjectToPlacement(obj)
+	if err != nil {
+		return err
+	}
 
 	// handle requeing for changes in placement, excluding deletion
 	if !isBeingDeleted(obj) {
+		// update placement resolution destinations since placement was updated
+		list, err := ocm.ListClustersBySelectors(c.ocmClient, placement.Spec.ClusterSelectors)
```

Comment:
"list" was never a good name for that variable, something more semantic would have been better. Now it is an even worse name, since the data type is no longer a list.

Reply:
> FindClustersBySelectors

Reply:
The variable is still named "list".